### PR TITLE
[IMPROVEMENT] Allow deployment of Prometheus ServiceMonitor with the Longhorn helm chart

### PR DESCRIPTION
**Pull request by yardenshoham**
When deploying Longhorn with helm, to enable metric collection we have to manually apply a ServiceMonitor object. This object is fairly simple so we add a value in the Longhorn helm chart to include it. The value is `false` by default.

- Closes https://github.com/longhorn/longhorn/issues/7041

### CODE REVIEW

**Review by innobead — APPROVED** [review]
LGTM.

**Comment by innobead** [comment]
@mantissahz Please help review and verify this PR before merging, and then drive the corresponding issue. Thanks.

**Comment by yardenshoham** [comment]
Usually, Prometheus watches all ServiceMonitors, regardless of namespace but I don't mind adding this change. I will do it in a few hours.

**Review by mantissahz — APPROVED** [review]
LGTM, @yardenshoham could you leave the message in the pending review and resolve it?

File: chart/templates/servicemonitor.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{{- if .Values.enableServiceMonitor }}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: longhorn-prometheus-servicemonitor
+  namespace: {{ include "release_namespace" . }}
```
[the remark this answers is not in the document]
Comment (yardenshoham) [reply chart/templates/servicemonitor.yaml:6]: Yes, as long as Prometheus watches all namespaces (which is the default behavior of kube-prometheus).

**Comment by yardenshoham** [comment]
https://github.com/longhorn/longhorn/pull/6157 is merged

**Pull request closed without merging**